src/topology/locally_constant/basic.lean

```diff
@@ -298,7 +298,10 @@ begin
   have := h x,
   set a := f x,
   set b := g x,
-  fin_cases a; fin_cases b; finish
+  fin_cases a; fin_cases b;
+  rw [h_1, h_2] at *,
```

Comment:
where do these hypothesis names come from?

Reply:
They're introduced by `fin_cases a` and `fin_cases b` respectively

Reply:
hmm, can we have `fin_cases a with ha` to avoid autogenerated names in proof scripts?

Reply:
`fin_cases a with ha` gives the error `unknown identifier 'ha'`.  I don't see anything in the documentation for how to name the introduced hypothesis.

Reply:
Ugh, `fin_cases` hardcodes the name `h`. `fin_cases with` does something different, it takes pexprs to change the shape of the cases instead of names, and it's also undocumented. Both of these need to be fixed.

Reply:
I moved it to a lemma in `data.fin.basic` and used `simp [forall_fin_two]` instead of `fin_cases`.

Reply:
Great, thank you